acme/dns_challenge.go

```diff
@@ -255,6 +255,20 @@ func FindZoneByFqdn(fqdn string, nameservers []string) (string, error) {
 
 		// Check if we got a SOA RR in the answer section
 		if in.Rcode == dns.RcodeSuccess {
+
+			// CNAME records cannot/should not exist at the root of a zone.
+			// So we skip a domain when a CNAME is found.
+			hasCNAME := false
+			for _, ans := range in.Answer {
+				if _, ok := ans.(*dns.CNAME); ok {
+					hasCNAME = true
```

Comment:
Would this look better if we had a label on the outer loop and would continue right here? Minor thing but what do you think?

Reply:
It might be even better to extract it into a small helper function.
I'll prepare a patch.